datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java

```diff
@@ -79,6 +126,107 @@ public static GraphQLEngine get() {
         return _engine;
     }
 
+    private static void configureQueryResolvers(final RuntimeWiring.Builder builder) {
+        builder.type("Query", typeWiring -> typeWiring
+                .dataFetcher("search", new AuthenticatedResolver<>(
+                        new SearchResolver(SEARCHABLE_TYPES)))
+                .dataFetcher("autoComplete", new AuthenticatedResolver<>(
+                        new AutoCompleteResolver(SEARCHABLE_TYPES)))
+                .dataFetcher("browse", new AuthenticatedResolver<>(
+                        new BrowseResolver(BROWSABLE_TYPES)))
+                .dataFetcher("browsePaths", new AuthenticatedResolver<>(
+                        new BrowsePathsResolver(BROWSABLE_TYPES)))
+                .dataFetcher("dataset", new AuthenticatedResolver<>(
+                        new LoadableTypeResolver<>(
+                                DATASET_TYPE,
+                                (env) -> env.getArgument(URN_FIELD_NAME))))
+                .dataFetcher("corpUser", new AuthenticatedResolver<>(
+                        new LoadableTypeResolver<>(
+                                CORP_USER_TYPE,
+                                (env) -> env.getArgument(URN_FIELD_NAME))))
+        );
+    }
+
+    /**
+     * Configures resolvers responsible for resolving the {@link com.linkedin.datahub.graphql.generated.Dataset} type.
+     */
+    private static void configureDatasetResolvers(final RuntimeWiring.Builder builder) {
+        builder
+            .type("Dataset", typeWiring -> typeWiring
+                    .dataFetcher("platform", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    DATA_PLATFORM_TYPE,
+                                    (env) -> ((Dataset) env.getSource()).getPlatform().getUrn()))
+                    )
+                    .dataFetcher("downstreamLineage", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    DOWNSTREAM_LINEAGE_TYPE,
+                                    (env) -> ((Dataset) env.getSource()).getUrn()))
+                    )
```

Comment:
shouldn't there be a corresponding upstream lineage type? On that note, should we just have a `lineage` type instead? Shouldn't down/up be a field?

Reply:
Interesting. So from a modeling perspective, you're suggesting something like an array of RelatedDataset objects wherein each says whether it is UP or DOWN of the source? That is something we can consider...

There is no upstream lineage type merely by the fact that calling the downstream `Datasets` endpoint returns the upstreamLineage (as it is persisted in our MySQL store), but does _not_ return a downstreamLineage field. Thus, we need to resolve it independently using the corresponding endpoint. That's what this resolver it for!

Reply:
The modeling question is an interesting one. When more lineage types come down the pipeline (Jobs, Dashboards), we'll likely want a more broad set of lineage that can be 1 of multiple types..

Reply:
Sounds good.